dbms/src/TableFunctions/TableFunctionRandom.cpp

```diff
@@ -0,0 +1,477 @@
+            throw Exception("Type 'Funclion' can not be stored in a table.", ErrorCodes::LOGICAL_ERROR);
+        case TypeIndex::AggregateFunction:
+            throw Exception("Random Generator not implemented for type 'AggregateFunction'.", ErrorCodes::NOT_IMPLEMENTED);
+        case TypeIndex::LowCardinality:
+            throw Exception("Random Generator not implemented for type 'LowCardinality'.", ErrorCodes::NOT_IMPLEMENTED);
+    }
+}
+
+StoragePtr TableFunctionRandom::executeImpl(const ASTPtr & ast_function, const Context & context, const std::string & table_name) const
+{
+    ASTs & args_func = ast_function->children;
+
+    if (args_func.size() != 1)
+        throw Exception("Table function '" + getName() + "' must have arguments.", ErrorCodes::LOGICAL_ERROR);
+
+    ASTs & args = args_func.at(0)->children;
+
+    if (args.size() < 2)
+        throw Exception("Table function '" + getName() + "' requires at least two arguments: "\
+                        " structure, limit(, max_array_length, max_string_length, random_seed).",
+                        ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    if (args.size() > 5)
+        throw Exception("Table function '" + getName() + "' requires at most five arguments: "\
+                        " structure, limit, max_array_length, max_string_length, random_seed.",
+                        ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    /// Parsing first argument as table structure and creating a sample block
+    std::string structure = args[0]->as<ASTLiteral &>().value.safeGet<String>();
+
+    UInt64 limit = 1;
+    UInt64 max_array_length = 10;
+    UInt64 max_string_length = 10;
+    UInt64 random_seed = 0; // zero for random
+
+    /// Parsing second argument if present
+    if (args.size() >= 2)
+        limit = args[1]->as<ASTLiteral &>().value.safeGet<UInt64>();
+
+    if (!limit)
+        throw Exception("Table function '" + getName() + "' limit should not be 0.", ErrorCodes::BAD_ARGUMENTS);
+
+    if (args.size() >= 3)
+        max_array_length = args[1]->as<ASTLiteral &>().value.safeGet<UInt64>();
+
+    if (args.size() >= 4)
+        max_string_length = args[1]->as<ASTLiteral &>().value.safeGet<UInt64>();
+
+    if (args.size() == 5)
+        random_seed = args[1]->as<ASTLiteral &>().value.safeGet<UInt64>();
+
+    ColumnsDescription columns = parseColumnsListFromString(structure, context);
+
+    Block res_block;
+    for (const auto & name_type : columns.getOrdinary())
+    {
+        MutableColumnPtr column = name_type.type->createColumn();
+        res_block.insert({std::move(column), name_type.type, name_type.name});
+    }
+
+    for (auto & ctn : res_block.getColumnsWithTypeAndName())
+    {
+        fillColumnWithRandomData(ctn.column->assumeMutableRef(), ctn.type, limit, max_array_length, max_string_length, random_seed);
+    }
+    auto res = StorageValues::create(StorageID(getDatabaseName(), table_name), columns, res_block);
```

Comment:
It won't work in streaming fashion.
We should be able to generate infinite amount of data for fuzz testing.

Reply:
StorageRandom and RandomSource should be created.

Reply:
added StorageGenerate